### ISSUE TITLE
Allow ability to specify language

### ISSUE THREAD

**Issue by Samreay** [mixn/carbon-now-cli#59]
Hi team, I notice in the settings, that you cannot manually specify the language.

```
	// setting this manually will be ignored
	l: 'auto',
```

Which is frustrating, because if you are trying to extract code snippets from a markdown file, or a Jupyter notebook, or any file which doesnt have the right extension, you do not get the correct formatting. It would be great if we could specify this manually, instead of my current solution (which is to ignore the ability to take snippets, write my own code that does the snippeting and saving to a temporary file before deleting said file). I get why you may not want it in the preset or config (because that should be language agnostic), but having it as a command line argument when invoking would be great! If its already there and Ive simply missed it when reading the docs, I apologise :)

**Comment by mixn** [comment]
Hey @Samreay, sorry for the late response, I somehow completely missed this — my bad! :) 

It [currently indeed will be ignored](https://github.com/mixn/carbon-now-cli/blob/63b7915a9df1a09949d2a5633fa10d987d666fce/src/helpers/default-settings.js#L7-L9) when set manually. 

I’m collecting feature requests, enhancements, bugs, etc. that will ship with `v2` of this CLI — the list is getting longer and longer though. :) But this will most likely be part of it since it’s quite straightforward to implement.

I Will update you on when `v2` is out — it’ll likely take a while still though. :) 

Thanks for opening and the idea!

**Comment by julianrubisch** [comment]
👋  do you accept PRs on this?

**Comment by feluelle** [comment]
+1 would love having this feature

**Comment by mixn** [comment]
So, I’m finalizing the big `v2` release (#83) and just wanted to implement this when I realized it might not be needed.

Specifying a language will work out of the box in v2 by using `"l": "YOUR_DESIRED_LANGUAGE"` e.g., `"l": "clojure"` inside one of your presets.

The question now is, is that enough? I feel like it’s definitely sufficient and solves this problem. The only “overhead” on the user’s end is that it is bound to a preset, aka. it won’t work *without* a preset with an associated `l`anguage key.

Example preset that will work with `v2` and use TypeScript as a language:

```
"verminal": {
    "t": "verminal",
    "bg": "#F30C96",
    "wt": "bw",
    "wc": true,
    "fm": "Anonymous Pro",
    "fs": "18px",
    "ln": false,
    "ds": false,
    "sl": "*",
    "dsyoff": "20px",
    "dsblur": "68px",
    "wa": true,
	"l": "application/typescript",
    "lh": "133%",
    "pv": "30px",
    "ph": "30px",
    "si": false,
    "wm": false,
    "es": "2x",
    "type": "png"
  }
```

If you’re unsure what language values Carbon’s site understands, they’re basically all values from [`extensions-map.helper.ts`](https://github.com/mixn/carbon-now-cli/blob/v2/src/helpers/cli/extensions-map.helper.ts).

This would also work with `--from-clipboard`, from `stdin`, etc.

Generally, I’m not a big fan of adding custom command line arguments like `--language` for something that is **100%** a Carbon option, not a `carbon-now-cli` option (as all existing arguments so far are).

This kind of opens up the gate to potentially adding more and more options that are solely relevant to Carbon itself, and I’d ideally like people to continue using presets for that.

If we end up adding `--language`, I’m scared people might start asking for basically everything to be configurable via command line arguments.

What are your thoughts on this? Thanks in advance!

@Samreay @julianrubisch @feluelle

**Comment by feluelle** [comment]
For my use-case the preset is sufficient. Thank you! :)

**Comment by mixn** [comment]
As I’m finalizing the huge [`v2` release](https://github.com/mixn/carbon-now-cli/issues/83), I’ve decided to run a small beta of it. 🙂 This issue is somehow covered in that release, so I’ll post this here. 🙂

To install the `v2` beta run

```
npm i carbon-now-cli@beta -g
```

[Docs 🚀 for `v2` can be found here.](https://github.com/mixn/carbon-now-cli/tree/v2/) 🙂 There are tons of new features and breaking changes, so make sure to re-read the docs if you’ve read them previously.

Happy testing and using! Looking forward to your feedback!
`v2` will officially drop any day now. 🥳 

P. S: I’m almost embarrassed about how long it took me to start working on `v2`, but at least once I did, I shipped it quite fast. 🙃 No more activity slumps in the future, though. 🤞
